conf/cassandra.yaml

```diff
@@ -1251,6 +1251,8 @@ server_encryption_options:
   # Set to a valid keystore if internode_encryption is dc, rack or all
   keystore: conf/.keystore
   keystore_password: cassandra
+  outbound_keystore: conf/.keystore
+  outbound_keystore_password: cassandra
   # Verify peer server certificates
   require_client_auth: false
```

Comment:
@jyothsnakonisa Do we have to make any changes to [OutboundConnectionInitiator](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/net/OutboundConnectionInitiator.java#L205) or [InboundConnectionInitiator](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/net/InboundConnectionInitiator.java#L534)?

Reply:
@jyothsnakonisa Have you thought of any other naming convention than `outbound`  here? `Outbound` is too contextual, meaning to whoever knows this code well it makes perfect sense but not otherwise - Ideally I wish we had named `server_encryption_options` as `internode_encryption_options` then we could have named this `client_keystore` and existing configuration as `server_keystore`. Just a thought based on my own confusion between Netty's Inbound/Outbound terminology +  Cassandra's Inbound/Outbound terminology.

Reply:
Also, I had a question on whether we can put two private keys in the same keystore - one for the server mode and the other for the client mode for internode connection. Would java be able to use the correct certificate in that case?

Reply:
We don't have to make any change in `OutboundConnectionInitiator` or `InboundConnectionInitiator` or `SSLFactory`. `AbstractSslContextFactory` has the logic to create appropriate keystore based on server connection or client connection.

Reply:
@maulin-vasavada  I think renaming `server_encryption_options` would be quite a big change and might need major version change. We are planning to get this patch in 4.0.

Reply:
I experimented with having two private keys in same keystore but that did not work as the first key that is trusted by the other party is picked during SSL handshake and we don't have control over which key to be picked.

Reply:
Thanks @jyothsnakonisa for running an experiment with two private keys in the same keystore. As I noted in one of the updates about - I am convinced about having a separate keystore for outbound for client auth. Also, I agree that renaming server_encryption_options would be lot of changes, I was just providing reasoning on why 'outbound' word is too contextual and what could be `ideal`. No changes expected on that from this PR from my side.

However, can you please take a look at one of the first comments in this series of comment? - @jyothsnakonisa Do we have to make any changes to [OutboundConnectionInitiator](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/net/OutboundConnectionInitiator.java#L205) or [InboundConnectionInitiator](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/net/InboundConnectionInitiator.java#L534) or here [SSLFactory](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/security/SSLFactory.java#L415)?

Reply:
Sorry @jyothsnakonisa I missed to see your comment about Oubound/Inbound Connection Initator and SSLFactory changes comment. Please resolve this.